src/GraphQL.Tests/Complexity/ComplexityMetaDataTests.cs

```diff
@@ -0,0 +1,149 @@
+using GraphQL.Execution;
+using GraphQL.MicrosoftDI;
+using GraphQL.Types;
+using GraphQL.Validation.Complexity;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.Tests.Complexity;
+
+public class ComplexityMetaDataTests : IClassFixture<ComplexityMetaDataFixture>
+{
+    private readonly ComplexityMetaDataFixture _fixture;
+
+    public ComplexityMetaDataTests(ComplexityMetaDataFixture fixture)
+    {
+        _fixture = fixture;
+    }
+
+    [Fact]
+    public async Task MetaData_ShouldBe_Used()
+    {
+        var result = await _fixture.AnalyzeAsync(@"
+query {
+    hero { #2
+        id #0
+        name #2
+        friends { #4
+            id #0
+            name #4
+            f1: friends { #12
+                name #12
+            }
+            f2: friends { #12
+                name #12
+            }
+        }
+    }
+}
+").ConfigureAwait(false);
+
+        result.Complexity.ShouldBe(60);
+        result.TotalQueryDepth.ShouldBe(4);
+    }
+
+    [Fact]
+    public async Task MetaData_With_Fragments_ShouldBe_Used()
+    {
+        var result = await _fixture.AnalyzeAsync(@"
+query {
+    hero { #2
+        id #0
+        name #2
+        ...friendsRoot #132(4/2*66)
+    }
+}
+
+fragment friendsRoot on Hero {
+    friends { #3
+        ...friendsIdName #27(9/2*6)
```

Comment:
I don't understand what's going on here, but it doesn't seem correct.  It should just be `3*6` - where 3 is the value of the parent and 6 is the total of the fragment.  Can we fix this while we are making changes here?

Reply:
Ok.  We can address it in a separate issue.  Thanks!

Reply:
#3164